akka-remote/src/main/scala/akka/remote/RemoteWatcher.scala

```diff
@@ -193,11 +193,11 @@ private[akka] class RemoteWatcher(
 
       // also watch from self, to be able to cleanup on termination of the watchee
       context watch watchee
-      watching += ((watchee, self))
+      watching += watchee → watching.get(watchee).fold(Set(self)) { _ + self }
```

Comment:
Not sure about speed argument. `Option.fold` is marked inline and only consist of one `if`
But I change to a `match`, it is more common usage